calibrators/marker_radar_lidar_calibrator/src/marker_radar_lidar_calibrator.cpp

```diff
@@ -1402,108 +1524,124 @@ void ExtrinsicReflectorBasedCalibrator::findCombinations(
-      total_crossval_calibrated_2d_distance_error / combinations.size();
-    double avg_crossval_calibrated_2d_yaw_error =
-      total_crossval_calibrated_2d_yaw_error / combinations.size();
-    output_metrics_.push_back(static_cast<float>(num_of_samples));
-    output_metrics_.push_back(static_cast<float>(avg_crossval_calibrated_2d_distance_error));
-    output_metrics_.push_back(static_cast<float>(avg_crossval_calibrated_2d_yaw_error));
-
-    double std_crossval_calibrated_2d_distance_error = calculate_std(
-      crossval_calibrated_2d_distance_error_vector, avg_crossval_calibrated_2d_distance_error);
-    double std_crossval_calibrated_2d_yaw_error =
-      calculate_std(crossval_calibrated_2d_yaw_error_vector, avg_crossval_calibrated_2d_yaw_error);
-    output_metrics_.push_back(static_cast<float>(std_crossval_calibrated_2d_distance_error));
-    output_metrics_.push_back(static_cast<float>(std_crossval_calibrated_2d_yaw_error));
+    Eigen::Isometry3d crossval_calibrated_radar_to_lidar_transformation =
+      crossval_estimator.getTransformation();
+    // calculate the error.
+    auto [crossval_calibrated_distance_error, crossval_calibrated_yaw_error] =
+      computeCalibrationError(crossval_calibrated_radar_to_lidar_transformation);
+
+    total_crossval_calibrated_distance_error += crossval_calibrated_distance_error;
+    total_crossval_calibrated_yaw_error += crossval_calibrated_yaw_error;
+    crossval_calibrated_distance_error_vector.push_back(crossval_calibrated_distance_error);
+    crossval_calibrated_yaw_error_vector.push_back(crossval_calibrated_yaw_error);
+  }
+
+  auto calculate_std = [](std::vector<double> & data, double mean) -> double {
+    double sum = 0.0;
+    for (size_t i = 0; i < data.size(); i++) {
+      sum += (data[i] - mean) * (data[i] - mean);
+    }
+    double variance = sum / data.size();
+    return std::sqrt(variance);
+  };
+
+  double avg_crossval_calibrated_distance_error =
+    total_crossval_calibrated_distance_error / combinations.size();
+  double avg_crossval_calibrated_yaw_error =
+    total_crossval_calibrated_yaw_error / combinations.size();
+  output_metrics_.push_back(static_cast<float>(num_of_samples));
+  output_metrics_.push_back(static_cast<float>(avg_crossval_calibrated_distance_error));
+  output_metrics_.push_back(static_cast<float>(avg_crossval_calibrated_yaw_error));
+
+  double std_crossval_calibrated_distance_error = calculate_std(
+    crossval_calibrated_distance_error_vector, avg_crossval_calibrated_distance_error);
+  double std_crossval_calibrated_yaw_error =
+    calculate_std(crossval_calibrated_yaw_error_vector, avg_crossval_calibrated_yaw_error);
+  output_metrics_.push_back(static_cast<float>(std_crossval_calibrated_distance_error));
+  output_metrics_.push_back(static_cast<float>(std_crossval_calibrated_yaw_error));
+}
+
+void ExtrinsicReflectorBasedCalibrator::crossValEvaluation()
+{
+  int tracks_size = static_cast<int>(converged_tracks_.size());
```

Comment:
can you change the parts that correspond to std::size?

Reply:
(this applies in several parts of the crossvalidation strategy )

Reply:
Addressed them in https://github.com/tier4/CalibrationTools/pull/180/commits/b9e8c9d0e3d1cc8afdb5fda5f29a6738e3e13ebb